packages/ts-moose-lib/src/commons.ts

```diff
@@ -118,6 +118,28 @@ export const RETRY_FACTOR_PRODUCER = 0.2;
 // Means all replicas need to acknowledge the message
 export const ACKs = -1;
 
+/**
+ * Creates the base producer configuration for Kafka.
+ * Used by both the SDK stream publishing and streaming function workers.
+ *
+ * @param maxMessageBytes - Optional max message size in bytes (synced with topic config)
+ * @returns Producer configuration object for the Confluent Kafka client
+ */
+export function createProducerConfig(maxMessageBytes?: number) {
+  return {
+    kafkaJS: {
+      idempotent: false, // Not needed for at-least-once delivery
+      acks: ACKs,
+      retry: {
+        retries: MAX_RETRIES_PRODUCER,
+        maxRetryTime: MAX_RETRY_TIME_MS,
+      },
+    },
+    "linger.ms": 0, // Send immediately - batching happens at application level
+    ...(maxMessageBytes && { "message.max.bytes": maxMessageBytes }),
```

Comment:
the logic for maxMessageBytes is kinda spread out

the way it currently is it will always be a number > 0 because of how maxMessageBytes is set in the call site at packages/ts-moose-lib/src/streaming-functions/runner.ts line 941

you could remove the `?` from maxMessageBytes in this function signature and then you don't need this object spread